include/ginkgo/core/base/composition.hpp

```diff
@@ -167,6 +167,60 @@ class Composition : public EnableLinOp<Composition<ValueType>>,
 };
 
 
+/**
+ * The UseComposition class can be used to store the composition information in
+ * LinOp.
+ *
+ * @tparam ValueType  precision of input and result vectors
+ */
+template <typename ValueType = default_precision>
+class UseComposition {
+public:
+    using value_type = ValueType;
+    /**
+     * Returns the composition opertor.
+     *
+     * @return composition
+     */
+    std::shared_ptr<Composition<ValueType>> get_composition() const
+    {
+        return composition_;
+    }
+
+    /**
+     * Returns the operator at index-th poistion of composition
+     *
+     * @return index-th operator
+     *
+     * @note when this composition is not set, this function always returns
+     *       nullptr. However, when this composition is set, it will throw
+     *       exception when exceeding index.
```

Comment:
I guess we are not always consistent with this, but maybe add an `@throws` doxygen comment then?